jumanji/environments/packing/tetris/env.py

```diff
@@ -0,0 +1,362 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Sequence, Tuple
+
+import chex
+import jax
+import jax.numpy as jnp
+import matplotlib
+from numpy.typing import NDArray
+
+from jumanji import specs
+from jumanji.env import Environment
+from jumanji.environments.packing.tetris import utils
+from jumanji.environments.packing.tetris.constants import (
+    NUM_ROTATIONS,
+    REWARD_LIST,
+    TETROMINOES_LIST,
+)
+from jumanji.environments.packing.tetris.types import Observation, State
+from jumanji.environments.packing.tetris.viewer import TetrisViewer
+from jumanji.types import TimeStep, restart, termination, transition
+from jumanji.viewer import Viewer
+
+
+class Tetris(Environment[State]):
+    """RL Environment for the game of Tetris.
+    The environment has a grid where the player can place tetrominoes.
+    The environment has the following characteristics:
+
+    - observation: `Observation`
+        - grid: jax array (int32) of shape (num_rows, num_cols)
+            representing the current state of the grid.
+        - tetromino: jax array (int32) of shape (4, 4)
+            representing the current tetromino sampled from the tetromino list.
+        - action_mask: jax array (bool) of shape (4,  num_cols).
+            For each tetromino there are 4 rotations, each one corresponds
+            to a line in the action_mask.
+            Mask of the joint action space: True if the action
+            (x_position and rotation degree) is feasible
+            for the current tetromino and grid state.
+    - action: multi discrete array of shape (2,)
+        - rotation_index: The degree index determines the rotation of the
+            tetromino: 0 corresponds to 0 degrees, 1 corresponds to 90 degrees,
+            2 corresponds to 180 degrees, and 3 corresponds to 270 degrees.
+        - x_position: int between 0 and num_cols - 1 (included).
+
+    - reward:
+        The reward is 0 if no lines was cleared by the action and a convex function of the number
+        of cleared lines otherwise.
+
+    - episode termination:
+        if the tetromino cannot be placed anymore (i.e., it hits the top of the grid).
+
+    ```python
+    from jumanji.environments import Tetris
+    env = Tetris()
+    key = jax.random.key(0)
+    state, timestep = jax.jit(env.reset)(key)
+    env.render(state)
+    action = env.action_spec().generate_value()
+    state, timestep = jax.jit(env.step)(state, action)
+    env.render(state)
+    ```
+    """
+
+    def __init__(
+        self,
+        num_rows: int = 10,
+        num_cols: int = 10,
+        time_limit: int = 4000,
+        viewer: Optional[Viewer[State]] = None,
+    ) -> None:
+        """Instantiates a `Tetris` environment.
+
+        Args:
+            num_rows: number of rows of the 2D grid. Defaults to 10.
+            num_cols: number of columns of the 2D grid. Defaults to 10.
```

Comment:
The image is 10x8. What setting do we want to go for? If we go for 10x10, we need to make the image and gif again. If we decide to go for 10x8, we need to change the default settings